insights/core/__init__.py

```diff
@@ -1289,34 +1293,67 @@ def raw_directory(self, directory):
 
 class AttributeDict(dict):
     """
-    Class to convert the access to each item in a dict as attribute.
+    Class to convert the access to the keys in ``fixed_attrs`` dictionary as
+    attributes.
 
     Examples:
         >>> data = {
         ... "fact1":"fact 1"
         ... "fact2":"fact 2"
         ... "fact3":"fact 3"
         ... }
-        >>> d_obj = AttributeDict(data)
-        {'fact1': 'fact 1', 'fact2': 'fact 2', 'fact3': 'fact 3'}
-        >>> d_obj['fact1']
-        'fact 1'
+        >>> d_obj = AttributeDict(data, fixed_attrs={'fact0': 'fact 0', 'fact1': ''})
+        {'fact0': 'fact 0', fact1': 'fact 1', 'fact2': 'fact 2', 'fact3': 'fact 3'}
+        >>> 'fact0' in d_obj
+        True
+        >>> d_obj['fact0']
+        'fact 0'
         >>> d_obj.get('fact1')
         'fact 1'
+        >>> d_obj.fact0
+        'fact 0'
         >>> d_obj.fact1
         'fact 1'
         >>> 'fact2' in d_obj
         True
-        >>> d_obj.get('fact3', default='no fact')
-        'fact 3'
-        >>> d_obj.get('fact4', default='no fact')
-        'no fact'
-    """
+        >>> hasattr(d_obj, 'fact2')
+        False
+
+    Arguments:
+        (dict): The base dictionary will work on
+        fixed_attrs (dict): The fixed keys should be specified in this directory
+            with the values in type :class:`type_info`
 
+    Raises:
+        TypeError: When more than one directories are passed
+        ValueError: When no or an empty 'fixed_attrs' is passed
+
+    """
+    type_info = namedtuple('type_info', field_names=['type', 'default'])
+    """namedtuple: Type for the values of the ``fixed_attrs`` dictionary"""
     def __init__(self, *args, **kwargs):
+        if len(args) > 1:
```

Comment:
@xiangce if args can only be 1, couldn't you just use a variable name for the parameter?

Reply:
That makes sense to me, will modify it. Thanks